fetcher/src/job/job.utils.ts

```diff
@@ -180,7 +180,18 @@ export function extractFeeds(
   const feeds = adapter.feeds.map((f) => {
     let proxy: IProxy
     try {
-      proxy = proxySelector(f.definition.url)
+      if (!f.definition.location) {
```

Comment:
It would be better to use the `selectProxyFn` from @bayram98, but now I noticed that it probably does not work as we expected since the function is called for every newly added feed. There is no reuse of mapping between subsequent. @bayram98 Please let me know if I am wrong 🙏 @nick-bisonai For now, we can keep it as it is.

Reply:
@martinkersner I think you are true. `selectProxyFn` is will be used for all newly added feeds, and will try to randomly and evenly distribute the proxies.  

I think there will be some problem with current approach, where feeds does not have `f.defination.location` will be still assigned the specific location proxy. Generally, `selectProxyFn` will assign the proxies from whole list of proxies.

Reply:
I am not sure we talk about the same issue, but I think `selectProxyFn` does not work now since there is no persistence in the mapping between calls. 🤔

Reply:
@martinkersner I think you might be true about the current `selectProxyFn` is not working as we expected before. I will test that later and will figure out.